webapi/src/test/scala/org/knora/webapi/responders/v2/ResourcesResponderV2Spec.scala

```diff
@@ -1161,31 +1179,85 @@ class ResourcesResponderV2Spec extends CoreSpec() with ImplicitSender {
-              dimY = 256
-            )
-          )
+      val inputResource = CreateImageMessage
+        .make(
+          resourceIri = resourceIri,
+          internalFilename = "IQUO3t1AABm-FSLC0vNvVpr.jp2",
+          dimX = 512,
+          dimY = 256,
+          resourceClassIri = "http://0.0.0.0:3333/ontology/0001/anything/v2#ThingPicture".toSmartIri
         )
+        .value
+
+      responderManager ! CreateResourceRequestV2(
+        createResource = inputResource,
+        featureFactoryConfig = defaultFeatureFactoryConfig,
+        requestingUser = anythingUserProfile,
+        apiRequestID = UUID.randomUUID
       )
 
-      val inputResource = CreateResourceV2(
-        resourceIri = Some(resourceIri.toSmartIri),
-        resourceClassIri = "http://0.0.0.0:3333/ontology/0001/anything/v2#ThingPicture".toSmartIri,
-        label = "test thing picture",
-        values = inputValues,
-        projectADM = SharedTestDataADM.anythingProject
+      expectMsgType[ReadResourcesSequenceV2](timeout)
+
+      // Get the resource from the triplestore and check it.
+
+      val outputResource = getResource(resourceIri, anythingUserProfile)
+
+      checkCreateResource(
+        inputResourceIri = resourceIri,
+        inputResource = inputResource,
+        outputResource = outputResource,
+        defaultResourcePermissions = defaultAnythingResourcePermissions,
+        defaultValuePermissions = defaultStillImageFileValuePermissions,
+        requestingUser = anythingUserProfile
       )
+    }
+
+    "create a resource with document representation" in {
+      // Create the resource.
+
+      val resourceIri: IRI = stringFormatter.makeRandomResourceIri(SharedTestDataADM.anythingProject.shortcode)
+
+      val inputResource = CreateDocumentMessage
+        .make(
+          resourceIri = resourceIri,
+          internalFilename = "IQUO3t1AABm-FSLC0vNvVpr.pdf"
+        )
+        .value
+
+      responderManager ! CreateResourceRequestV2(
+        createResource = inputResource,
+        featureFactoryConfig = defaultFeatureFactoryConfig,
+        requestingUser = anythingUserProfile,
+        apiRequestID = UUID.randomUUID
+      )
+
+      expectMsgType[ReadResourcesSequenceV2](timeout)
+
+      // Get the resource from the triplestore and check it.
+
+      val outputResource = getResource(resourceIri, anythingUserProfile)
+
+      checkCreateResource(
+        inputResourceIri = resourceIri,
+        inputResource = inputResource,
+        outputResource = outputResource,
+        defaultResourcePermissions = defaultAnythingResourcePermissions,
+        defaultValuePermissions = defaultStillImageFileValuePermissions,
+        requestingUser = anythingUserProfile
+      )
+    }
+
+    "create a resource with bundle representation" in {
+      // Create the resource.
+
+      val resourceIri: IRI = stringFormatter.makeRandomResourceIri(SharedTestDataADM.anythingProject.shortcode)
+
+      val inputResource = CreateDocumentMessage
```

Comment:
Shouldn't this be a `CreateBundleMessage`?

Reply:
you're right. scary that this even works...